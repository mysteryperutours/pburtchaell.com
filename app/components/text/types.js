export const HEADER_1 = 'h1';
export const HEADER_2 = 'h2';
export const HEADER_3 = 'h3';
export const BODY = 'p';
