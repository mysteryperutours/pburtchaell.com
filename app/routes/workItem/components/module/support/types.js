const types: Object = {
  MODULE_MEDIA: 'module-media',
  MODULE_TEXT: 'module-text',
  MODULE_MEDIA_FULL_WIDTH: 'module-media-full-width',
  MODULE_MEDIA_LEFT: 'module-media-left',
  MODULE_MEDIA_RIGHT: 'module-media-right'
};

export default types;
