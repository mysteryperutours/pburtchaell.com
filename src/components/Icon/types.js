export const GITHUB = 'GITHUB';
export const TWITTER = 'TWITTER';
export const FACEBOOK = 'FACEBOOK';
export const DRIBBBLE = 'DRIBBBLE';
