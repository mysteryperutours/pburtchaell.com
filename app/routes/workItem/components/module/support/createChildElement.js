import { createElement, Element } from 'react';
import Text from '../../../../../components/text';
import Image from '../../../../../components/image';

// Get the child element type
function getChildElementType(type) {
  switch (type) {
    case 'text':
      return Text;
    case 'image':
      return Image;
    default:
      return 'div';
  }
}

/**
 * @function createChildElement
 * @description Create a new element for a child descried as a data object.
 */
function createChildElement(child): Element<*> {
  const { children } = child;

  return createElement(
    getChildElementType(child.type),
    {
      ...(child.meta ? child.meta : {})
    },
    Array.isArray(children) ? children.map(createChildElement) : children
  );
}

export default createChildElement;
