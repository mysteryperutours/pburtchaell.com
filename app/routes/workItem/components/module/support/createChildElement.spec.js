import { Children } from 'react';
import createChildElement from './createChildElement';
import types from './types';

describe(createChildElement.name, () => {
  it('should return a single child element of type Text', () => {
    const element = createChildElement({
      type: 'text',
      children: 'Foo'
    });

    expect(mount(element)).toMatchSnapshot();
  });

  it('should return a single child element of type Image', () => {
    const element = createChildElement({
      type: 'image',
      meta: {
        src: 'https://path/to/image.png',
        alt: 'Foo',
        height: 500,
        width: 700
      }
    });

    expect(mount(element)).toMatchSnapshot();
  });
});
