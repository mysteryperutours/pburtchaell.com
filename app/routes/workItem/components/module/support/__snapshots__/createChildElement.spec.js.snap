exports[`createChildElement should return a single child element of type Image 1`] = `
<Image>
  <div
    className="image-preload-container"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }>
    <img
      className="image-preload"
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "maxHeight": "100%",
          "maxWidth": "100%",
        }
      } />
  </div>
</Image>
`;

exports[`createChildElement should return a single child element of type Text 1`] = `
<Text
  type="p">
  <p>
    Foo
  </p>
</Text>
`;
